File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        
        int[] answer = new int[targets.length];
        
        // for(int i = 0; i<answer.length; i++){
        //     answer[i] = -1;
        // }
        
        
        for(int i =0; i<targets.length; i++){
            String target = targets[i];
            int count = 0;

            for(int j = 0; j<target.length(); j++){
            
                String str = String.valueOf(target.charAt(j));
                
                int min = Integer.MAX_VALUE; //int 타입 최대값

                for (String x : keymap) {
                    int idx = x.indexOf(str);
                    if (idx != -1) {
                        min = Math.min(min, idx + 1);
                    }
                }
                
                if(min == Integer.MAX_VALUE){ //수가 없다는 뜻
                    count = -1;
                    break;
                }
                count += min;
            }
            answer[i] = count;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    
    let answer = [];
    
    for(let target of targets){
        
        let count = 0;
        
        for(let key of target.split("")){
            let min = 1000000000;
            
            for(let board of keymap){
                if(board.includes(key)){
                    min = Math.min(min, board.indexOf(key)+1); //keymap 중에서 가장 작은 번호의 자판만 가져옴
                }
            }
            
            if(min === 1000000000){ //min값이 변하지 않았다는 건 keymap에 자판이 존재하지 않음
                count = -1;
                break;
            }
        
            count+= min;
        }
        
        answer.push(count);

    }
    
    return answer
}